docs/src/interface.md

```diff
@@ -1,22 +1,35 @@
 # The AbstractSciMLOperator Interface
 
-## Formal Properties of DiffEqOperators
+## Formal Properties of SciMLOperators
 
 These are the formal properties that an `AbstractSciMLOperator` should obey
 for it to work in the solvers.
 
-## AbstractDiffEqOperator Interface Description
+1. An `AbstractSciMLOperator` represents a linear or nonlinear operator with input/output
+   being `AbstractArray`s. Specifically, a SciMLOperator, `L`, of size `(M,N)` accepts
+   input argument `u` with leading length `N`, i.e. `size(u, 1) == N`, and returns an
+   `AbstractArray` of the same dimension with leading length `M`, i.e. `size(L * u, 1) == M`.
+   Internally, `L` lazily reshapes `u` to a matrix of size `(N, length(u) \div N)` and
+   independently acts on the column-vectors. The reshape operation is skipped for
+   `AbstractVecOrMat` arguments.
+2. SciMLOperators can be applied to an `AbstractArray` via overloaded `Base.*`, or
+   the in-place `LinearAlgebra.mul!`. Additionally, operators are allowed to be time,
+   or parameter dependent. The state of a SciMLOperator can be updated by calling
+   the mutating function `update_coefficients!(L, u, p, t)` where `p` representes
+   parameters, and `t`, time.  Calling a SciMLOperator as `L(du, u, p, t)` or out-of-place
+   `L(u, p, t)` will automatically update the state of `L` before applying it to `u`.
```

Comment:
or simply, `L(u,p,t)` is the operation `L(u,p,t) * u` as is common in the definition of semilinear ODE systems.